lib/dal/src/workspace_snapshot/migrator/v2.rs

```diff
@@ -0,0 +1,92 @@
+use petgraph::{prelude::*, visit::IntoEdgeReferences};
+use si_events::ulid::Ulid;
+use std::collections::{HashMap, HashSet};
+
+use crate::{
+    workspace_snapshot::{
+        content_address::ContentAddress,
+        graph::{deprecated::DeprecatedWorkspaceSnapshotGraphV1, LineageId},
+        node_weight::NodeWeight,
+    },
+    EdgeWeight, WorkspaceSnapshotGraphV2,
+};
+
+use super::SnapshotGraphMigratorResult;
+
+pub fn migrate_v1_to_v2(
```

Comment:
In a future change we might want to consider wrapping an `#[instrument]` around this one in order to have some "upgrade cost" info. Kind of nice that it's in one place